src/ai/PathFinder.cpp

```diff
@@ -387,7 +387,7 @@ PathFinder::NodeId PathFinder::getNearestNode(const Vec3f & pos) const {
 	
 	for(size_t i = 0; i < map_s; i++) {
 		float dist = glm::distance2(map_d[i].pos, pos);
-		if(dist < distance && map_d[i].nblinked) {
+		if((dist < distance && map_d[i].nblinked) && !(map_d[i].flags & ANCHOR_FLAG_BLOCKED)) {
```

Comment:
This will most likely have an impact on how exactly do the NPCs wander around to look for the player (getNearestNode() is used in lookFor()). Instead of plotting paths to hopeless nodes the NPCs will head for reachable ones. Optically, this doesn't seem to make a difference, as they still wander around cluelessly.